gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineNullAway.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.plugins;
+
+import java.util.Optional;
+import net.ltgt.gradle.errorprone.ErrorProneOptions;
+import org.gradle.api.Action;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.plugins.ExtensionAware;
+import org.gradle.api.tasks.compile.JavaCompile;
+
+public final class BaselineNullAway implements Plugin<Project> {
+
+    private static final Logger log = Logging.getLogger(BaselineNullAway.class);
+
+    @Override
+    public void apply(Project project) {
+        project.getPluginManager().withPlugin("com.palantir.baseline-error-prone", _unused0 -> {
+            project.getPluginManager().withPlugin("java", _unused1 -> {
+                applyToProject(project);
+            });
+        });
+    }
+
+    private void applyToProject(Project project) {
+        String version = Optional.ofNullable(BaselineNullAway.class.getPackage().getImplementationVersion())
+                .orElseGet(() -> {
+                    log.warn("BaselineNullAway is using 'latest.release' - "
+                            + "beware this compromises build reproducibility");
+                    return "latest.release";
+                });
+        project.getDependencies().add("errorprone", "com.palantir.baseline:baseline-null-away:" + version);
+        configureErrorProneOptions(project, new Action<ErrorProneOptions>() {
+            @Override
+            public void execute(ErrorProneOptions options) {
+                options.option("NullAway:AnnotatedPackages", "com.palantir");
```

Comment:
I guess we need to provide a default package to make this work out of the box per https://github.com/uber/NullAway/wiki/Configuration#annotated-packages so this seems reasonable, though any non-Palantir projects would need to override this to enable NullAway for their code.

It would be nice to include some other commonly used packages like `com.google.common` for Guava but that may require more opt-in work for folks adopting this from the get-go per the "Nullness annotations" bits from [Guava 31.0 release](https://github.com/google/guava/releases/tag/v31.0) specifically the NullAway bits mentioning https://github.com/google/guava/issues/6126#issue-1325269284

Reply:
Happy to add guava by default for now, it's easier to remove packages than add them.

I'd considered introspecting the maven coordinate group, and adding something like the first two segments. So `net.ckozak.foo` would result in `net.ckozak` as an annotated package. However, I think there are sufficiently many edge cases that I'd prefer to punt solving this until we need to do so.

Reply:
I kicked off https://github.com/palantir/tritium/compare/ds/guava-nullaway?expand=1 as a quick check last night and that [build failed](https://app.circleci.com/jobs/github/palantir/tritium/19000) so I think adding Guava here might cause more false positives than desired by default, so we probably do want just `options.option("NullAway:AnnotatedPackages", "com.palantir");` for now. Sorry for the run around on that.

Reply:
No worries, I appreciate that you called out the idea! I had just tested a snapshot on tritium and found the same thing :-)

Reply:
https://github.com/palantir/tritium/pull/1570 to fix up tritium's one flagged issue